internal/pkg/config/v1/config.go

```diff
@@ -250,10 +275,40 @@ func (c *Config) Save() error {
 	c.restoreOverwrittenKafka(tempKafka)
 	c.restoreOverwrittenAuthToken(tempAuthToken)
 	c.restoreOverwrittenAuthRefreshToken(tempAuthRefreshToken)
+	c.restoreOverwrittenCredentials(tempAPISecrets)
 
 	return nil
 }
 
+func (c *Config) encryptCredentialsAPISecret(tempAPISecrets map[string]string) error {
```

Comment:
Looks like we're doing more than just encrypting the API secret

Reply:
it's generating salt and nonce if nil and then encrypt the secret

Reply:
Let's give this function a more general name to reflect that?

Reply:
I think it's expected to generate nonce and salt during encryption though. What's your suggestion?

Reply:
Oh, I didn't fully understand what's going on here. I thought we were in a different function. Can we move the nonce/salt generation to a separate function so we don't have duplicated code?

Reply:
refactored into a separate function to generate salt and nonce!